### PR TITLE
[#8354] [Platform][API] Audit controller error handling.

### PR DESCRIPTION
**Pull request by mahendranbhat**
**Description:**

Use the new global exception handling in the audit controller.

**Test Plan:**

Its mostly refactor. Unit tests are updated.
Run server and test basic operations.

sbt test :

```	at play.db.ebean.DefaultEbeanConfig$EbeanConfigParser$WrappingDatasource.getConnection(DefaultEbeanConfig.java:153)
	at io.ebeaninternal.server.transaction.TransactionFactoryBasic.createQueryTransaction(TransactionFactoryBasic.java:28)
	... 18 common frames omitted
[info] Passed: Total 1321, Failed 0, Errors 0, Passed 1317, Skipped 4
[success] Total time: 835 s (13:55), completed 25 May, 2021 12:48:13 PM``

### CODE REVIEW

File: managed/src/main/java/com/yugabyte/yw/controllers/AuditController.java [hunk 1]
```diff
@@ -24,63 +24,39 @@
...
   /**
    * GET endpoint for getting the user associated with a task.
    * @return JSON response with the corresponding audit entry.
    */
   public Result getTaskAudit(UUID customerUUID, UUID taskUUID) {
-    Customer customer = Customer.get(customerUUID);
-    if (customer == null) {
-      return ApiResponse.error(BAD_REQUEST, "Invalid Customer UUID: " + customerUUID);
+    Customer.getOrBadRequest(customerUUID);
+    Audit entry = Audit.getFromTaskUUID(taskUUID);
```
Comment (sb-yb) [inline managed/src/main/java/com/yugabyte/yw/controllers/AuditController.java:40]: Audit.getOrBadRequest(customerUUID, taskUUID) ->
This should lookup by BOTH customer and task and throw BAD_REQUEST if not found.

File: managed/src/main/java/com/yugabyte/yw/controllers/AuditController.java [hunk 2]
```diff
@@ -24,63 +24,39 @@
...
+        String.format("Task %s does not belong to customer %s", taskUUID, customerUUID));
     }
   }
 
   public Result getUserFromTask(UUID customerUUID, UUID taskUUID) {
-    Customer customer = Customer.get(customerUUID);
-    if (customer == null) {
-      return ApiResponse.error(BAD_REQUEST, "Invalid Customer UUID: " + customerUUID);
+    Customer.getOrBadRequest(customerUUID);
+    Audit entry = Audit.getFromTaskUUID(taskUUID);
```
Comment (sb-yb) [inline managed/src/main/java/com/yugabyte/yw/controllers/AuditController.java:52]: Same. Easiest way to see that you need more cleanup is the fact that there is duplicate code between two methods.

**Review by sb-yb — CHANGES_REQUESTED** [review]

File: managed/src/main/java/com/yugabyte/yw/models/Audit.java [hunk 3]
```diff
@@ -155,6 +158,15 @@ public static Audit getFromTaskUUID(UUID taskUUID) {
     return find.query().where().eq("task_uuid", taskUUID).findOne();
   }
 
+  public static Audit getOrBadRequest(UUID customerUUID, UUID taskUUID) {
+    Audit entry = find.query().where().eq("task_uuid", taskUUID).findOne();
```
Comment (sb-yb) [inline managed/src/main/java/com/yugabyte/yw/models/Audit.java:162]: you have to lookup by both `customerUUID` and `taskUUID`

**Review by sb-yb — CHANGES_REQUESTED** [review]

File: managed/src/main/java/com/yugabyte/yw/controllers/AuditController.java [hunk 4]
```diff
@@ -24,63 +24,27 @@
...
-        return ApiResponse.success(user);
-      }
-      else {
-        return ApiResponse.error(BAD_REQUEST,
-          String.format("Task %s does not belong to customer %s", taskUUID, customerUUID));
-      }
-    } catch (Exception e) {
-      return ApiResponse.error(INTERNAL_SERVER_ERROR, "Unable to fetch user.");
-    }
+    Customer.getOrBadRequest(customerUUID);
```
Comment (sb-yb) [inline managed/src/main/java/com/yugabyte/yw/controllers/AuditController.java:45]: OMG!!! so much codes deletes !!!

**Review by sb-yb — APPROVED** [review]

File: managed/src/main/java/com/yugabyte/yw/controllers/AuditController.java (hunk 4 above)
Comment (mahendranbhat) [reply managed/src/main/java/com/yugabyte/yw/controllers/AuditController.java:45]: Yeah, now the controller is clean and takes 5 sec to check what it is exactly doing without much distraction!!

File: managed/src/main/java/com/yugabyte/yw/controllers/EncryptionAtRestController.java [hunk 5]
```diff
@@ -36,65 +36,59 @@
...
+    Customer customer = Customer.getOrBadRequest(customerUUID);
+    try {
+      TaskType taskType = TaskType.CreateKMSConfig;
+      ObjectNode formData = (ObjectNode) request().body().asJson();
+      KMSConfigTaskParams taskParams = new KMSConfigTaskParams();
+      taskParams.kmsProvider = Enum.valueOf(KeyProvider.class, keyProvider);
+      taskParams.providerConfig = formData;
+      taskParams.customerUUID = customerUUID;
+      taskParams.kmsConfigName = formData.get("name").asText();
+      formData.remove("name");
```
Comment (sb-yb) [inline managed/src/main/java/com/yugabyte/yw/controllers/EncryptionAtRestController.java:59]: try catch should end here. Do not include line 60. Errors beyond this point are not due to bad request

File: managed/src/main/java/com/yugabyte/yw/controllers/EncryptionAtRestController.java [hunk 6]
```diff
@@ -130,120 +124,108 @@ public Result listKMSConfigs(UUID customerUUID) {
...
+            "Deleting KMS configuration %s for customer %s",
+            configUUID.toString(), customerUUID.toString()));
+    Customer customer = Customer.getOrBadRequest(customerUUID);
+    try {
+      KmsConfig config = KmsConfig.get(configUUID);
+      TaskType taskType = TaskType.DeleteKMSConfig;
+      KMSConfigTaskParams taskParams = new KMSConfigTaskParams();
+      taskParams.kmsProvider = config.keyProvider;
+      taskParams.customerUUID = customerUUID;
+      taskParams.configUUID = configUUID;
```
Comment (sb-yb) [inline managed/src/main/java/com/yugabyte/yw/controllers/EncryptionAtRestController.java:139]: same catch here.

**Review by sb-yb — CHANGES_REQUESTED** [review]

File: managed/src/main/java/com/yugabyte/yw/models/Audit.java [hunk 7]
```diff
@@ -134,6 +137,17 @@ public static Audit getFromTaskUUID(UUID taskUUID) {
     return find.query().where().eq("task_uuid", taskUUID).findOne();
   }
 
+  public static Audit getOrBadRequest(UUID customerUUID, UUID taskUUID) {
+    Customer.getOrBadRequest(customerUUID);
+    Audit entry =
+        find.query().where().eq("task_uuid", taskUUID).eq("customerUUID", customerUUID).findOne();
```
Comment (sb-yb) [inline managed/src/main/java/com/yugabyte/yw/models/Audit.java:143]: Does this work? `customerUUID` or `customer_uuid`? Not sure ebeans is smart enough to mask the difference.

**Review by sb-yb — CHANGES_REQUESTED** [review]

**Review by sb-yb — APPROVED** [review]

**Pull request merged**